apps/url-shortener/main.go

```
@@ -228,6 +248,39 @@ func (s *Server) handler(w http.ResponseWriter, r *http.Request) {
 	renderHTML(w, r, tmpl, pageVariables)
 }
 
+type UpdateRequest struct {
+	Name   string `json:"name"`
+	Active bool   `json:"active"`
+}
+
+func (s *Server) togglehandler(w http.ResponseWriter, r *http.Request) {
+	var data UpdateRequest
+	if r.Method == http.MethodPost {
```

This function should return an error if non `POST` method is invoked on `/api` endpoint

---

Done